src/aoc.rs

```diff
@@ -317,3 +320,103 @@ pub fn read(
     println!("\n{}", from_read(desc.as_bytes(), col_width));
     Ok(())
 }
+
+fn get_private_leaderboard_results(
+    args: &Args,
+    session: &str,
+    leaderboard: &str,
+    year: PuzzleYear,
+) -> AocResult<PrivateLeaderboard> {
+    debug!("🦌 Fetching private leaderboard {}", leaderboard);
+
+    let url = format!(
+        "https://adventofcode.com/{}/leaderboard/private/view/{}.json",
+        year, leaderboard
+    );
+
+    let leaderboard: PrivateLeaderboard =
+        build_client(session, "application/json")?
+            .get(&url)
+            .send()
+            .and_then(|response| response.error_for_status())
+            .and_then(|response| response.json())
+            .map_err(AocError::from)?;
+    Ok(leaderboard)
+}
+
+pub fn show_private_leaderboard_results(
+    args: &Args,
+    session: &str,
+    leaderboard: &str,
+) -> AocResult<()> {
+    let (year, day) = puzzle_year_day(args.year, args.day)?;
```

Comment:
I think instead of calling `puzzle_year_day` we need a new function. Something that: 1) unwraps the year if given or returns the latest event year, e.g. `let year = args.year.unwrap_or_else(latest_event_year);` and 2) returns the last unlocked day for the selected year.